### ISSUE TITLE
Identity Server Account Registration API failing

### ISSUE THREAD

**Issue by DeadlyHollows** [vector-im/element-web#16085]
### Description

Hi,

I have setup a synapse server and when I try to invite a user via email, I get an Internal Server Error (from both the identity servers https://vector.im and https://matrix.org)

### Steps to reproduce

When I try to invite a user using their email id, I get this error -

![image](https://user-images.githubusercontent.com/22254056/103853025-268c9100-50d3-11eb-8387-1122368bc601.png)

The network request send to the identity server (vector.im / matrix.org) returns:

{"errcode": "M_UNKNOWN", "error": "Internal Server Error"}

I expect the server to return back a token as mentioned [here](https://matrix.org/docs/spec/identity_service/latest#post-matrix-identity-v2-account-register)

### Version information

- **Platform**: web (in-browser)
- **Browser**: Firefox 84.0.2 (64-bit)
- **OS**: Windows
- **URL**: Running it on a private server. Element web version **v1.7.16**

**Comment by turt2live** [comment]
is your homeserver federated? It should pass the tests at https://federationtester.matrix.org/ - if not, this error is expected.

**Comment by DeadlyHollows** [comment]
Hi @turt2live,

Really appreciate your prompt response on this. I had made changes to my setup but still failing short of 1 check (**MatchingServerName**).

I had set the certificate for my domain and things are working correctly when the endpoint is requested using the domain name:
![image](https://user-images.githubusercontent.com/22254056/103891326-a59fba80-510f-11eb-9d47-41158e5a3403.png)

But the Matrix Federation Tester makes use of IP address and there my applied certificate doesn't works.

What can I do in such scenario? Any help would be appreciated.

Thank you!
